src/pymor/algorithms/timestepping.py

```diff
@@ -10,227 +10,619 @@
+    interpolation
+        Type of temporal interpolation to be used. Currently implemented are: piecewise constant (P0) and piecewise
+        linear (P1).
     """
 
-    def __init__(self, nt):
+    IteratorType = ExplicitEulerIterator
+
+    def __init__(self, nt, num_values=None, interpolation='P1'):
+        super().__init__(num_values, interpolation)
+
+        assert isinstance(nt, Number)
+        assert nt > 0
+
         self.__auto_init(locals())
 
-    def solve(self, initial_time, end_time, initial_data, operator, rhs=None, mass=None, mu=None, num_values=None):
-        if mass is not None:
-            raise NotImplementedError
-        return explicit_euler(operator, rhs, initial_data, initial_time, end_time, self.nt, mu, num_values)
 
+class ExplicitRungeKuttaIterator(SingleStepTimeStepperIterator):
+
+    def __init__(self, stepper, t0, t1, U0, A, F, M, mu, return_iter, return_times):
+        super().__init__(stepper, t0, t1, U0, A, F, M, mu, return_iter, return_times)
+        self.dt = dt = (t1 - t0) / stepper.nt
+        # use the ones from base, these are checked and converted in super().__init__()
+        A, F, M, mu = self.A, self.F, self.M, self.mu
+
+        # prepare the function f in d_t y = f(t, y)
+        if not isinstance(M, IdentityOperator) and not _depends_on_time(M, mu):
+            M = M.assemble(mu)
+        if not _depends_on_time(A, mu):
+            A = A.assemble(mu)
+
+        if isinstance(F, ZeroOperator):
+            if isinstance(M, IdentityOperator):
+                def f(t, y):
+                    mu_t = mu.with_(t=t)
+                    return -1*A.apply(y, mu=mu_t)
+            else:
+                def f(t, y):
+                    mu_t = mu.with_(t=t)
+                    return M.apply_inverse(-1*A.apply(y, mu=mu_t), mu=mu_t, initial_guess=y)
+        else:
+            if not _depends_on_time(F, mu):
+                F = F.assemble(mu)
+            if isinstance(M, IdentityOperator):
+                def f(t, y):
+                    mu_t = mu.with_(t=t)
+                    return F.as_vector(mu=mu_t) - A.apply(y, mu=mu_t)
+            else:
+                def f(t, y):
+                    mu_t = mu.with_(t=t) or Mu({'t': t})
+                    return M.apply_inverse(F.as_vector(mu=mu_t) - A.apply(y, mu=mu_t), mu=mu_t, initial_guess=y)
+
+        self.f = f
+
+    def _step_function(self, U_n, t_n):
+        c, A, b = self.stepper.butcher_tableau
+        # compute stages
+        s = len(c)
+        stages = U_n.space.empty(reserve=s)
+        for j in range(s):
+            t_n_j = t_n + self.dt*c[j]
+            U_n_j = U_n.copy()
+            for l in range(j):
+                U_n_j.axpy(self.dt*A[j][l], stages[l])
+            U_n_j = self.f(t_n_j, U_n_j)
+            stages.append(U_n_j, remove_from_other=True)
+        # compute step
+        U_np1 = U_n.copy()
+        for j in range(s):
+            U_np1.axpy(self.dt*b[j], stages[j])
```

Comment:
```suggestion
        stages = U_n.space.empty(reserve=s+1)
        stages.append(U_n)
        for j in range(s):
            t_n_j = t_n + self.dt*c[j]
            U_n_j = stages.lincomb([1] + self.dt*A[j][:j])
            U_n_j = self.f(t_n_j, U_n_j)
            stages.append(U_n_j, remove_from_other=True)
        # compute step
        U_np1 = stages.lincomb([1] + self.dt*b)
```

Probably buggy, but shouldn't something like this work?